burner-wallet-plugin/tokenbridge-bw-exchange/src/burner-wallet/assets/BridgeableERC20Asset.ts

```diff
@@ -0,0 +1,48 @@
+import { ERC20Asset } from '@burner-wallet/assets'
+import { MEDIATOR_ABI } from '../../utils'
+import { toBN } from 'web3-utils'
+
+interface BridgeableERC20Constructor {
+  abi?: object
+  address: string
+  id: string
+  name: string
+  network: string
+  bridgeAddress: string
+}
+
+export default class BridgeableERC20Asset extends ERC20Asset {
+  protected bridgeAddress: string
+  private _bridge
+
+  constructor({ bridgeAddress, ...params }: BridgeableERC20Constructor) {
+    super({ ...params })
+    this.bridgeAddress = bridgeAddress.toLowerCase()
+  }
+
+  getBridgeContract() {
+    if (!this._bridge) {
+      const Contract = this.getWeb3().eth.Contract
+      this._bridge = new Contract(MEDIATOR_ABI, this.bridgeAddress)
+    }
+    return this._bridge
+  }
+
+  async _send({ from, to, value }) {
+    if (to.toLowerCase() === this.bridgeAddress) {
+      const allowance = await this.allowance(from, to)
+      if (toBN(allowance).lt(value)) {
```

Comment:
While testing, I found that the `approve` method was never called even if the allowance was `0`. Converting `value` to BN will properly compare the two numbers and call `approve` if required
```suggestion
      if (toBN(allowance).lt(toBN(value))) {
```

Reply:
You are right. Thanks